### PR TITLE
 MCR-2827 MCRConcurrentHashMap.get wrong behavior

### PR DESCRIPTION
**Pull request by toKrause**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2827).

### CODE REVIEW

File: mycore-base/src/test/java/org/mycore/common/config/MCRConfigurationTest.java [hunk 1]
```diff
@@ -18,131 +18,156 @@
...
+
+    private Pair<String, String> createConflictConfiguration() {
+
         char[] letters = "abcdefghijklmnopqrstuvwxyz".toCharArray();
 
         int limit = 50; // initial size of char arrays
 
-        char[][] className = new char[2][limit];
-        char[][] property = new char[2][limit];
+        String classNameA = MCRConfigurationSingletonCollisionClassA.class.getName();;
```
Comment (Hammer1279) [inline mycore-base/src/test/java/org/mycore/common/config/MCRConfigurationTest.java]:
```suggestion
        String classNameA = MCRConfigurationSingletonCollisionClassA.class.getName();
```

File: mycore-base/src/test/java/org/mycore/common/config/MCRConfigurationTest.java [hunk 2]
```diff
@@ -18,131 +18,156 @@
...
-            if (value.matches(".+MCRConfigurationSingletonCollisionClass[AB]")) {
-                assertNotNull(MCRConfiguration2.instanceHolder.get(new MCRConfiguration2.SingletonKey(key, value)));
-            }
-        });
+    private int spread(int h) {
+        return (h ^ (h >>> 16)) & 0x7fffffff /* all usable bits */;
     }
+
+    private static String toSuitableString(char[] value) {
+        return String.valueOf(value).trim().replaceAll(Pattern.quote(" "), "");
```
Comment (Hammer1279) [inline mycore-base/src/test/java/org/mycore/common/config/MCRConfigurationTest.java]: Is the `Pattern.quote()` really necessary?

Comment (toKrause) [reply mycore-base/src/test/java/org/mycore/common/config/MCRConfigurationTest.java]: I think it's best practice to always use `Pattern.quote` when using a string literal as a pattern.

THB, I don't really see why the the `replaceAll` is necessary at all, but it's in the original code. If necessary, a `replaceAll("\\s+", "")` would probably be best?

File: mycore-base/src/test/java/org/mycore/common/config/singletoncollision/MCRConfigurationSingletonCollisionClassA.java [hunk 3]
```diff
@@ -1,36 +1,34 @@
...
+ * (at your option) any later version.
+ *
+ * MyCoRe is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MyCoRe.  If not, see <http://www.gnu.org/licenses/>.
+ */
```
Comment (Hammer1279) [inline mycore-base/src/test/java/org/mycore/common/config/singletoncollision/MCRConfigurationSingletonCollisionClassA.java:17]: Note for later: this has to also be done on class B

**Review by Hammer1279 — APPROVED** [review]
Looks good to me.

**Pull request merged**